### PR TITLE
[squid:S1854] Dead stores should be removed

### PR DESCRIPTION
**Pull request by AymanDF**
This pull request is focused on resolving occurrences of Sonar rule 
squid:S1854 - “ Dead stores should be removed ”. 

You can find more information about the issue here: 
https://dev.eclipse.org/sonar/rules/show/squid:S1854

Please let me know if you have any questions.
Ayman Abdelghany.

### CODE REVIEW

**Comment by sankarbhavanib** [comment]
@AymanDF  Thank you for pull request , we will review it and get back to you.

File: src/main/java/com/paypal/core/SDKUtil.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ private SDKUtil() {
 	 * @return Replaced text
 	 */
 	public static String escapeInvalidXmlChars(String textContent) {
-		StringBuilder stringBuilder = null;
+		StringBuilder stringBuilder;
 		String response = null;
```
Comment (sankarbhavanib) [inline src/main/java/com/paypal/core/SDKUtil.java]: can we remove null for response String also

File: src/main/java/com/paypal/core/SDKUtil.java [hunk 2]
```diff
@@ -220,7 +220,7 @@ public static String escapeInvalidXmlCharsRegex(Boolean boolContent) {
 	 */
 	public static String escapeInvalidXmlCharsRegex(Double doubleContent) {
 		String response = null;
```
Comment (sankarbhavanib) [inline src/main/java/com/paypal/core/SDKUtil.java]: Can you please remove null for response String also

File: src/main/java/com/paypal/core/SDKUtil.java [hunk 3]
```diff
@@ -238,7 +238,7 @@ public static String escapeInvalidXmlCharsRegex(Double doubleContent) {
 	 */
 	public static String escapeInvalidXmlCharsRegex(Float floatContent) {
 		String response = null;
```
Comment (sankarbhavanib) [inline src/main/java/com/paypal/core/SDKUtil.java]: Can you please remove null for response String also

File: src/main/java/com/paypal/core/rest/PayPalResource.java [hunk 4]
```diff
@@ -225,10 +225,10 @@ public static String getLastResponse() {
 			HttpMethod httpMethod, String resourcePath, String payLoad,
 			Class<T> clazz) throws PayPalRESTException {
 		T t = null;
```
Comment (sankarbhavanib) [inline src/main/java/com/paypal/core/rest/PayPalResource.java]: Can you please remove null for t

Comment (sankarbhavanib) [inline src/main/java/com/paypal/core/rest/PayPalResource.java]: Can you please remove null for t

Comment (sankarbhavanib) [inline src/main/java/com/paypal/core/rest/PayPalResource.java]: also can you please do same change at https://github.com/DevFactory/sdk-core-java/blob/0a8c7d955f90a416992d498df25bedb11d684267/src/main/java/com/paypal/core/rest/PayPalResource.java#L382

**Comment by sankarbhavanib** [comment]
👍

File: src/main/java/com/paypal/sdk/util/OAuthSignature.java [hunk 5]
```diff
@@ -247,7 +247,7 @@ public boolean verifyV1Signature(String signature) throws OAuthException {
 	 * @throws OAuthException
 	 */
 	private String normalizeURI(String uri) throws OAuthException {
-		String normalizedURI = "", port = "", scheme = "", path = "", authority = "";
+		String normalizedURI, port = "", scheme, path = "", authority;
```
Comment (jaypatel512) [inline src/main/java/com/paypal/sdk/util/OAuthSignature.java]: This probably needs to be something like this:
```java
String normalizedURI, port, scheme, path, authority;
```

**Comment by jaypatel512** [comment]
Hey @AymanDF !

According to [Java Language Specifications](http://docs.oracle.com/javase/specs/jls/se7/jls7.pdf), Sec 4.12.5 it mentions that

> A local variable (§14.4, §14.14) must be explicitly given a value before it is
used, by either initialization (§14.4) or assignment (§15.26), in a way that can be
verified using the rules for definite assignment (§16).

I am not sure if your pull request obeys that. Can you please confirm ?

**Comment by AymanDF** [comment]
Hey Jay,

Yes, I confirm it obeys the mentioned specifications.

**Pull request merged**